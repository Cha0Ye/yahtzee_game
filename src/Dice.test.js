import React from 'react';
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Dice from './Dice';

// smoke test
it('renders without crashing', () => {
    const toggleLockedFn = jest.fn();
    mount(<Dice dice={Array.from({ length: 5 })} 
                locked={Array(5).fill(false)}
                handleClick={toggleLockedFn}/>);
});

// snapshot test unlocked
it("matches snapshot for unlocked", function() {
    const toggleLockedFn = jest.fn();
    let wrapper = mount(<Dice dice={Array.from({ length: 5 })} 
                locked={[true, false, false, false, false]}
                handleClick={toggleLockedFn}/>);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});
