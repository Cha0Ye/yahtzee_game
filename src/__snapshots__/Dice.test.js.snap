// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for unlocked 1`] = `
<Dice
  dice={
    Array [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
  handleClick={[Function]}
  locked={
    Array [
      true,
      false,
      false,
      false,
      false,
    ]
  }
>
  <div
    className="Dice"
  >
    <Die
      handleClick={[Function]}
      idx={0}
      key="0"
      locked={true}
    >
      <button
        className="Die Die-locked"
        onClick={[Function]}
      />
    </Die>
    <Die
      handleClick={[Function]}
      idx={1}
      key="1"
      locked={false}
    >
      <button
        className="Die"
        onClick={[Function]}
      />
    </Die>
    <Die
      handleClick={[Function]}
      idx={2}
      key="2"
      locked={false}
    >
      <button
        className="Die"
        onClick={[Function]}
      />
    </Die>
    <Die
      handleClick={[Function]}
      idx={3}
      key="3"
      locked={false}
    >
      <button
        className="Die"
        onClick={[Function]}
      />
    </Die>
    <Die
      handleClick={[Function]}
      idx={4}
      key="4"
      locked={false}
    >
      <button
        className="Die"
        onClick={[Function]}
      />
    </Die>
  </div>
</Dice>
`;
